src/llir/abi.rs

```diff
@@ -225,39 +240,68 @@ fn arg_encoding_from_attrs(param: &abi_ast::Param, emitter: &dyn Emitter) -> Res
 }
 
 fn int_from_attrs(param: &abi_ast::Param, emitter: &dyn Emitter) -> Result<Option<ArgEncoding>, ErrorReported> {
-    let (size, default_ty_color) = match param.format_char.value {
+    let (size, unsigned, mut hex_radix, default_ty_color) = match param.format_char.value {
         // FIXME: Uu should be unsigned but I'm not sure yet if I want  `i(signed)`, `i(unsigned)`, or `i(sign=1)`
-        'S' => (4u8, None),
-        's' => (2, None),
-        'U' => (4, None),
-        'u' => (2, None),
-        'n' => (4, Some(TypeColor::Enum(auto_enum_names::anm_sprite()))),
-        'N' => (4, Some(TypeColor::Enum(auto_enum_names::anm_script()))),
-        'E' => (4, Some(TypeColor::Enum(auto_enum_names::ecl_sub()))),
+        'S' => (4u8, false, false, None),
+        's' => (2u8, false, false, None),
+        'c' => (1u8, false, false, None),
+        'U' => (4u8, true, false, None),
+        'u' => (2u8, true, false, None),
+        'b' => (1u8, true, false, None),
+        'n' => (4u8, false, false, Some(TypeColor::Enum(auto_enum_names::anm_sprite()))),
+        'N' => (4u8, false, false, Some(TypeColor::Enum(auto_enum_names::anm_script()))),
+        'E' => (4u8, false, false, Some(TypeColor::Enum(auto_enum_names::ecl_sub()))),
+        'C' => (4u8, true, true, None),
         _ => return Ok(None),  // not an integer
     };
 
     param.clone().deserialize_attributes(emitter, |de| {
         let user_ty_color = de.accept_value("enum")?.map(|ident| TypeColor::Enum(ident.value));
         let arg0 = de.accept_flag("arg0")?;
+        let imm = de.accept_flag("imm")?;
+        let is_hex = de.accept_flag("hex")?;
 
         if let Some(arg0_flag) = arg0 {
-            if size != 2 {
+            if (size - 1) >= 2 {
                 return Err(emitter.as_sized().emit(error!(
-                    message("timeline arg0 must be word-sized ('s' or 'u')"),
+                    message("timeline arg0 must be word-sized or less ('s', 'u', 'c', or 'b')"),
                     primary(arg0_flag, ""),
                 )));
             }
         }
+        
+        if is_hex.is_some() {
+            hex_radix = true;
+        }
```

Comment:
```suggestion
        let radix = match is_hex.is_some() {
            true => ast::IntRadix::Hex,
            false => ast::IntRadix::Dec,
        };
```

Reply:
This is structured specifically to allow color format to default to hex now that it's been merged with the integer formats. Otherwise the radix would be forced to decimal unless you specify C(hex), which would defeat the point.

Reply:
```suggestion
        if is_hex.is_some() {
            hex_radix = true;
        }
        let radix = match hex_radix {
            true => ast::IntRadix::Hex,
            false => ast::IntRadix::Dec,
        };
```
I just wanted to pull the radix match out of the bigger struct to reduce duplication in case we add more radixes.